parachain/integration-test/src/lib.rs

```diff
@@ -0,0 +1,469 @@
+		.await
+		.map_err(|err| anyhow!("StateMachineUpdated stream subscription failed: {err:?}"))?;
+
+	let mut block_height_correction = chain_a_client.query_finalized_height().await?;
+	while let Some(update_event_a) = state_machine_a_update_stream.next().await {
+		match update_event_a {
+			Ok(event_a) => {
+				// As it tracks finalized height some blocks will be skipped in the stream and they
+				// may contain the response
+				block_height_correction = block_height_correction
+					.checked_sub(event_a.latest_height)
+					.map_or(event_a.latest_height, |rem| {
+						if rem > 0 {
+							block_height_correction + 1
+						} else {
+							event_a.latest_height
+						}
+					});
+
+				let block_hash =
+					client_a.rpc().block_hash(Some(block_height_correction.into())).await?.unwrap();
+				let fetched_events =
+					client_a.events().at(block_hash).await?.find_first::<GetResponse>()?;
+
+				match fetched_events {
+					Some(res_event) => {
+						response_event = Some(res_event.clone());
+						break
+					},
+					None => continue,
+				}
+			},
+			Err(err) => {
+				panic!("Error in state machine update stream: {err:?}")
+			},
+		}
+	}
+
+	let encoded_value = match response_event.unwrap().0[0].clone() {
+		Some(value) => value,
+		None => {
+			panic!("Value not found")
+		},
+	};
+
+	Ok(encoded_value)
+}
+
+/// Configure the state machines and spawn the messaging relayer
+async fn create_clients(
+) -> Result<(SubstrateClient<Hyperbridge>, SubstrateClient<Hyperbridge>), anyhow::Error> {
+	let chain_a_config = SubstrateConfig {
+		state_machine: StateMachine::Kusama(2000),
+		hashing: None,
+		consensus_state_id: Some("PARA".to_string()),
+		rpc_ws: "ws://127.0.0.1:9990".to_string(), // url from local-testnet zombienet config
+		max_rpc_payload_size: None,
+		signer: Some(
+			"0xe5be9a5092b81bca64be81d212e7f2f9eba183bb7a90954f7b76361f6edb5c0a".to_string(),
+		),
+		latest_height: None,
+		max_concurent_queries: None,
+	};
+
+	let chain_b_config = SubstrateConfig {
+		state_machine: StateMachine::Kusama(2001),
+		hashing: None,
+		consensus_state_id: Some("PARA".to_string()),
+		rpc_ws: "ws://127.0.0.1:9991".to_string(),
+		max_rpc_payload_size: None,
+		signer: Some(
+			"0xe5be9a5092b81bca64be81d212e7f2f9eba183bb7a90954f7b76361f6edb5c0a".to_string(),
+		),
+		latest_height: None,
+		max_concurent_queries: None,
+	};
+
+	// setup state machines
+	let chain_a_sub_client =
+		SubstrateClient::<KeccakSubstrateChain>::new(chain_a_config.clone()).await?;
+	let chain_b_sub_client =
+		SubstrateClient::<KeccakSubstrateChain>::new(chain_b_config.clone()).await?;
+
+	Ok((chain_a_sub_client, chain_b_sub_client))
+}
+
+/// Assertion is on ismp related events, and state changes on source and destination chain
+/// Alice in chain A sends 100_000 tokens to Alice in chain B
+#[tokio::test(flavor = "multi_thread")]
+#[ignore]
+async fn submit_transfer_function_works() -> Result<(), anyhow::Error> {
```

Comment:
Use a better name here
```suggestion
async fn submit_transfer_function_works() -> Result<(), anyhow::Error> {
```